Backpack/NavigationBar/Classes/BPKNavigationBar.m

```diff
@@ -0,0 +1,244 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKNavigationBar.h"
+
+#import <Backpack/Common.h>
+#import <Backpack/Color.h>
+#import <Backpack/Label.h>
+
+#import "BPKNavigationBarLargeTitleView.h"
+#import "BPKNavigationBarTitleView.h"
+
+
+const CGFloat BPKNavigationBarExpandedFullHeight = 96;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface BPKNavigationBar()
+// Title views
+@property(nonatomic, strong, readonly) BPKNavigationBarLargeTitleView *largeTitleView;
```

Comment:
@k0nserv just to know, there is some reason to make these properties `readonly` if they are in the .m file?

Reply:
I got inspired because I was writing some Swift with lazy computed properties and did the same thing for Objective-C. I really like this pattern, but I am open to being challenged by yourself or @georgegillams on it

Reply:
It's not a bad approach at all but I was wondering, since these properties are only in the .m and not in the .h file is the `readonly` necessary and therefore the `synthesize` too? 
Because in terms of visibility annotation the properties are not reachable by any other class that wants to use this `BPKNavigationBar` right?

Reply:
I feel like `readonly` is clearer in intent and I know I tend to make mistakes. In this case

```objective-c
self.largeTitleView = [....];
```

would be a mistake so it's nice that this prevents that from happening